cmd/etcd/delete.go

```diff
@@ -0,0 +1,101 @@
+package etcd
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+	"github.com/wal-g/wal-g/utility"
+)
+
+var confirmed = false
+
+// deleteCmd represents the delete command
+var deleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: internal.DeleteShortDescription,
+}
+
+var deleteBeforeCmd = &cobra.Command{
+	Use:     internal.DeleteBeforeUsageExample, // TODO : improve description
+	Example: internal.DeleteBeforeExamples,
+	Args:    internal.DeleteBeforeArgsValidator,
+	Run:     runDeleteBefore,
+}
+
+var deleteRetainCmd = &cobra.Command{
+	Use:       internal.DeleteRetainUsageExample, // TODO : improve description
+	Example:   internal.DeleteRetainExamples,
+	ValidArgs: internal.StringModifiers,
+	Args:      internal.DeleteRetainArgsValidator,
+	Run:       runDeleteRetain,
+}
+
+var deleteEverythingCmd = &cobra.Command{
+	Use:       internal.DeleteEverythingUsageExample,
+	Example:   internal.DeleteEverythingExamples,
+	ValidArgs: internal.StringModifiersDeleteEverything,
+	Args:      internal.DeleteEverythingArgsValidator,
+	Run:       runDeleteEverything,
+}
+
+func runDeleteBefore(cmd *cobra.Command, args []string) {
+	storage, err := internal.ConfigureStorage()
+	tracelog.ErrorLogger.FatalOnError(err)
+
+	deleteHandler, err := newEtcdDeleteHandler(storage.RootFolder())
+	tracelog.ErrorLogger.FatalOnError(err)
+
+	deleteHandler.HandleDeleteBefore(args, confirmed)
+}
+
+func runDeleteRetain(cmd *cobra.Command, args []string) {
+	storage, err := internal.ConfigureStorage()
+	tracelog.ErrorLogger.FatalOnError(err)
+
+	deleteHandler, err := newEtcdDeleteHandler(storage.RootFolder())
+	tracelog.ErrorLogger.FatalOnError(err)
+
+	deleteHandler.HandleDeleteRetain(args, confirmed)
+}
+
+func runDeleteEverything(cmd *cobra.Command, args []string) {
+	storage, err := internal.ConfigureStorage()
+	tracelog.ErrorLogger.FatalOnError(err)
+
+	deleteHandler, err := newEtcdDeleteHandler(storage.RootFolder())
+	tracelog.ErrorLogger.FatalOnError(err)
+
+	deleteHandler.DeleteEverything(confirmed)
+}
+
+func init() {
+	cmd.AddCommand(deleteCmd)
+	deleteCmd.AddCommand(deleteBeforeCmd, deleteRetainCmd, deleteEverythingCmd)
+	deleteCmd.PersistentFlags().BoolVar(&confirmed, internal.ConfirmFlag, false, "Confirms backup deletion")
+}
+
+func newEtcdDeleteHandler(folder storage.Folder) (*internal.DeleteHandler, error) {
```

Comment:
Maybe we should place DeleteHandler with other handlers (internal/databases/etcd)

Reply:
something like that https://github.com/wal-g/wal-g/blob/master/internal/databases/mysql/delete.go